openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIHeaderMapper.java

```diff
@@ -98,32 +112,36 @@
      * Assign header values to OAS header parameter.
      */
     private void completeHeaderParameter(List<Parameter> parameters, String headerName, HeaderParameter headerParameter,
-                                         StringSchema stringSchema, NodeList<AnnotationNode> annotations, Node node) {
+                                         Schema<?> headerSchema, NodeList<AnnotationNode> annotations, Node node) {
 
         if (!annotations.isEmpty()) {
             AnnotationNode annotationNode = annotations.get(0);
             headerName = getHeaderName(headerName, annotationNode);
         }
         if (node instanceof ArrayTypeDescriptorNode) {
             ArrayTypeDescriptorNode arrayNode = (ArrayTypeDescriptorNode) node;
-            if (arrayNode.memberTypeDesc().kind() == SyntaxKind.STRING_TYPE_DESC) {
-                ArraySchema arraySchema = new ArraySchema();
-                arraySchema.setItems(stringSchema);
-                headerParameter.schema(arraySchema);
-                headerParameter.setName(headerName);
-                parameters.add(headerParameter);
+            ArraySchema arraySchema = new ArraySchema();
+            SyntaxKind kind = arrayNode.memberTypeDesc().kind();
+            String item = kind.toString().split(UNDERSCORE)[0].trim().toLowerCase(Locale.ENGLISH);
```

Comment:
Shall we improve this logic and have a set of if conditions checking for each kind, instead of deriving it from the syntax kind string to be more future proof?

Reply:
Yeah, I also thought to use `if` for checking each type of kind and feel a little bit of chaos within the code, then I went with this process. I will try to improve the logic as much as possible :)